patterns/factorymethod/shipyards/src/main/java/com/scottlogic/dp/factorymethod/shipyards/game/Shipyard.java

```diff
@@ -5,20 +5,24 @@
     private int metal = 100;
     private int cloth = 100;
 
-    private final FishingShipBlueprints blueprints = new FishingShipBlueprints();
+    private ShipBlueprints blueprints = new FishingShipBlueprints();
 
-    FishingBarge constructShip() {
+    Ship constructShip() {
         MaterialRequirements requirements = blueprints.materialNeeded();
         Material material = prepareMaterial(requirements);
 
-        FishingBarge ship = blueprints.fishingShip(material);
+        Ship ship = blueprints.fishingShip(material);
         return ship;
     }
 
     public String showOffer() {
         return String.format("We can create [%s]", blueprints.shipType());
     }
 
+    public void upgrade() {
+        blueprints = new SturdyFishingShipBlueprints();
+    }
```

Comment:
This is good! Nothing wrong here. I guess another task would be to make Blueprints come in the upgrade() method. In the same manner, we could create the shipyard with blueprints passed in the constructor. That way the Shipyard only knows about the blueprints but not the concrete ones.

Which pattern that does that remind you of? :)